### PR TITLE
Feature/create job

### PR DESCRIPTION
**Pull request by AmeerAmjed**
Handel :- 
1- Create function **createJob** in **api service**.
2- Create function createJob in **repository** and implement in **RepositoryImpl**.
3- Create usecase **CreateJobUseCase**.
4- Create state  **CreateJobUiState** and **JobFormUiState**.
5- Create CreateJobViewModel.

### CODE REVIEW

File: app/src/main/java/com/cupcake/jobsfinder/domain/useCase/CreateJobUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.cupcake.jobsfinder.domain.useCase
+
+import com.cupcake.jobsfinder.data.remote.response.JobDto
+import com.cupcake.jobsfinder.data.repository.RepositoryImpl
+import javax.inject.Inject
+
+class CreateJobUseCase @Inject constructor(
+    private val repository: RepositoryImpl
```
Comment (msalarab) [inline app/src/main/java/com/cupcake/jobsfinder/domain/useCase/CreateJobUseCase.kt]: use repository interface

File: app/src/main/java/com/cupcake/jobsfinder/domain/useCase/CreateJobUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+) {
+    suspend operator fun invoke(paramCreateJob: ParamJobInfo): Boolean {
+        return repository.createJob(paramCreateJob.toJobDto())
+    }
+
+    data class ParamJobInfo(
```
Comment (msalarab) [inline app/src/main/java/com/cupcake/jobsfinder/domain/useCase/CreateJobUseCase.kt:15]: put it in model folder

File: app/src/main/java/com/cupcake/jobsfinder/ui/create_job/CreateJobViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class CreateJobViewModel @Inject constructor(
+    private val createJob: CreateJobUseCase
+) : BaseViewModel() {
+
+    private val _jobUiState = MutableStateFlow(CreateJobUiState())
+    val jobUiState = _jobUiState.asStateFlow()
```
Comment (msalarab) [inline app/src/main/java/com/cupcake/jobsfinder/ui/create_job/CreateJobViewModel.kt:18]: Why are you casting here?

**Review by msalarab — CHANGES_REQUESTED** [review]
Edit the first review 🗡️

**Review by msalarab — APPROVED** [review]

**Review by msalarab — APPROVED** [review]

**Review by TarekIdrees — APPROVED** [review]

**Pull request merged**